src/history/hs.c

```diff
@@ -484,6 +473,9 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MUL
     btree_id = btree->id;
     __wt_modify_vector_init(session, &modifies);
 
+    /* Disable bulk loads into history store. */
+    WT_WITH_BTREE(session, ((WT_CURSOR_BTREE *)cursor)->btree, __wt_cursor_disable_bulk(session));
```

Comment:
We might reach here and eventually not make an insert into the history store for various reasons (eg: all updates are obsolete), if you want to disable bulk load as close to the first insert as possible, you can explore moving this statement into ```__hs_insert_record_with_btree```.

Other than this change looks good.

Reply:
I don't think in this instance it matter which function we put it in, all we're doing is moving the time at which its called outside of config. I suppose its more correct to put it in  __hs_insert_record_with_btree, so I'll move it there.